keps/sig-scheduling/2891-simplified-config/README.md

```diff
@@ -0,0 +1,861 @@
+
+However these "behind-the-scenes" endpoints must still be configured in the `KubeSchedulerConfiguration`
+provided by the user for every plugin that uses them. For the average user this is an unnecessary
+and confusing step -- they only care that the plugin works, not how it works, and so
+internal processing steps are usually irrelevant to them.
+
+Therefore, a way for users to simply toggle all of a plugin's extension points
+(or a subset of the extension points as defined necessary by the plugin developer)
+would remove complexity and the potential for misconfiguration.
+
+### Goals
+
+* Provide an option for simplified plugin config from the end-user's perspective
+* Enable this config in all in-tree default scheduler plugins
+
+### Non-Goals
+
+* Refactor or introduce any breaking changes to the scheduling framework
+
+## Proposal
+
+A new extension point, `MultiPoint` (final name TBD) will be made available to
+users and developers which handles plugin registration for multiple extension
+points simultaneously.
+
+From the developer's perspective, this extension point will be an optional interface to
+implement in their plugin's code. This approach has several benefits:
+
+* **Does not break existing plugins**: Developers will not need to react to this change in
+order for their plugins to be configurable
+* **Code organization**: A specific extension point provides an accepted pattern
+and precedent to follow keeping the plugin ecosystem consistent, while also simplifying
+necessary documentation and examples needed from the scheduling maintainers.
+* **Consistent with other extension points**: This design can follow the established
+pattern for extension points within the scheduling framework and meet all of the definitions
+therein. This makes it easy to understand within the existing constructs of the framework.
+
+From the user's perspective, this will be a new (optional) field in the scheduler's
+config to enable or disable plugins across multiple extension points. It will be up to
+the user to know if the plugin they are using implements the new extension point, however
+in-tree plugins can guarantee to implement it and the existing framework error handling
+should promptly alert users for any third-party plugins which do not implement it.
+
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user, I wish to enable plugin `XYZ` for all applicable extension points (ie, `Filter` 
+and `Score`). I do not care about internal implementation or extension points such as 
+`PreFilter`, nor do I want to have to update my config should plugin `XYZ` change its 
+implementation within the scheduling framework to use different extension points.
+
+#### Story 2
+
+As a user, I wish to enable plugin `XYZ` for only *some* of its applicable extension 
+points (ie `Filter` but *not* `Score`). I still do not care about extension points such 
+as `PreFilter`. I want to have control over which subsets of extension points are enabled.
```

Comment:
Can you include the sample YAML to achieve each of these stories?

Reply:
For this story (subsets) the sample YAML is dependent on which implementation we choose. Option 1 defines allows subset disabling through plugin args while option 2 would use the existing regular extension points

Reply:
It sounds like Option 2 would detach the user intent from what the developer provides. That is a big benefit for out-of-tree plugins. It also reduces effort for us (kube-scheduler) to maintain disablement arguments for each plugin.